Test.Activities.Automation.ActivityLib/Models/Activity/Services/SyncActivityService.cs

```diff
@@ -0,0 +1,246 @@
+                if (spListRootMentors == null) throw new Exception("Getting SP root mentor list failed");
+
+                var spActivities = new List<SpActivity>();
+
+                var dateRangeQuery = new SPQuery
+                {
+                    Query =
+                        $"<Where><And><Geq><FieldRef Name=\"Date\"/><Value Type=\"DateTime\">{minDate:yyyy-MM-dd}</Value></Geq><Leq><FieldRef Name=\"Date\"/><Value Type=\"DateTime\">{maxDate:yyyy-MM-dd}</Value></Leq></And></Where>"
+                };
+
+                foreach (var item in spListActivities.GetItems(dateRangeQuery).Cast<SPListItem>())
+                {
+                    var mentor = SPHelper.GetLookUpUserValue(spListMentors, item, Constants.Activity.Mentor, Constants.Activity.Employee);
+                    var mentorValue = SPHelper.GetLookUpItemId(spListMentors, item, Constants.Activity.Mentor,
+                        Constants.Activity.Employee);
+
+                    var rootMentor = SPHelper.GetLookUpUserValue(spListRootMentors, item, Constants.Activity.RootMentor, Constants.Activity.Employee);
+                    var rootMentorValue = SPHelper.GetLookUpItemId(spListRootMentors, item, Constants.Activity.RootMentor, Constants.Activity.Employee);
+
+                    var month = SPHelper.GetIntValue(item, Constants.Activity.Month);
+                    var year = SPHelper.GetIntValue(item, Constants.Activity.Year);
+
+                    var activities = SPHelper.GetMultiChoiceValue(item, Constants.Activity.Activities);
+                    var paths = SPHelper.GetMultiChoiceValue(item, Constants.Activity.Paths);
+
+                    spActivities.Add(new SpActivity
+                    {
+                        RootMentorId = rootMentor?.User.ID,
+                        RootMentorLookupId = rootMentorValue,
+                        MentorId = mentor?.User.ID,
+                        MentorLookupId = mentorValue,
+                        Month = month,
+                        Year = year,
+                        Activities = new List<string>(activities),
+                        Paths = new List<string>(paths),
+                        Id = item.ID,
+                    });
+                }
+
+                return spActivities;
+            }
+            catch (Exception e)
+            {
+                throw new Exception($"Getting existing activities from SP failed. {e.Message}");
+            }
+        }
+
+        void UpdateSpActivities(IEnumerable<SpActivity> itemsToUpdate, SPWeb web)
+        {
+            try
+            {
+                _logger?.LogInformation("Updating SP activities list");
+
+                var spList = web.Lists.TryGetList(Constants.Lists.Activities);
+                if (spList == null) throw new Exception("Getting SP list failed");
+
+                foreach (var item in itemsToUpdate)
+                {
+                    if (item.IsNew)
```

Comment:
You may move this logic and `InsertActivity` and `UpdateActivity` methods to new method of `SpActivity` class